flink-runtime/src/main/java/org/apache/flink/runtime/state/RegisteredStateMetaInfoUtils.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.state;
+
+import org.apache.flink.api.common.state.v2.StateDescriptor;
+import org.apache.flink.api.common.typeutils.TypeSerializerSnapshot;
+import org.apache.flink.api.common.typeutils.base.ListSerializer;
+import org.apache.flink.api.common.typeutils.base.MapSerializer;
+import org.apache.flink.runtime.state.metainfo.StateMetaInfoSnapshot;
+import org.apache.flink.util.Preconditions;
+
+/**
+ * Utilities for transforming {@link RegisteredKeyValueStateBackendMetaInfo} to {@link
+ * org.apache.flink.runtime.state.v2.RegisteredKeyValueStateBackendMetaInfo}.
+ */
+public class RegisteredStateMetaInfoUtils {
+
+    private RegisteredStateMetaInfoUtils() {}
+
+    public static org.apache.flink.runtime.state.v2.RegisteredKeyValueStateBackendMetaInfo
+            createMetaInfoV2FromV1Snapshot(StateMetaInfoSnapshot stateMeta) {
+
+        Preconditions.checkArgument(
+                stateMeta
+                        .getBackendStateType()
+                        .equals(StateMetaInfoSnapshot.BackendStateType.KEY_VALUE));
+
+        StateDescriptor.Type newStateType;
+        TypeSerializerSnapshot oldValueSerializerSnapshot =
+                stateMeta.getTypeSerializerSnapshot(
+                        StateMetaInfoSnapshot.CommonSerializerKeys.VALUE_SERIALIZER);
+        TypeSerializerSnapshot newUserKeySerializerSnapshot = oldValueSerializerSnapshot;
+        TypeSerializerSnapshot newStateSerializerSnapshot = null;
```

Comment:
```
newUserKeySerializerSnapshot = null;
newStateSerializerSnapshot = oldValueSerializerSnapshot?
```

Otherwise, the `NPE` will be thrown, we can add tests about other states in `ForStAsyncAndSyncCompatibilityTest` to verify.